Upgrade prometheus image to v2.19.3

yktsubo opened the pull request:
- Make e2e prometheus test consistent by using querying target metadata API

This PR updates prometheus image to v2.19.3 so that we can get basename without cutting out the metrics name in our e2e test. New codes using querying target metadata API.
https://prometheus.io/docs/prometheus/latest/querying/api/#querying-target-metadata

However, this is still experimental so I won't merge if it's a risk to use the experimental API.

antoninbas commented on test/e2e/prometheus_test.go:
```diff
@@ -254,8 +256,8 @@ func testMetricsFromPrometheusServer(t *testing.T, data *TestData, prometheusJob
-	path := url.PathEscape("match[]={job=\"" + prometheusJob + "\"}")
-	queryUrl := fmt.Sprintf("http://%s:%d/api/v1/series?%s", hostIP, nodePort, path)
+	path := url.PathEscape("match_target={job=\"" + prometheusJob + "\"}")
+	queryUrl := fmt.Sprintf("http://%s:%d/api/v1/targets/metadata?%s", hostIP, nodePort, path)
```
please add a comment to explain that this API (`GET /api/v1/targets/metadata`) has been available since Prometheus v2.4.0, but is still considered experimental and may break in the future.

yktsubo replied:
Added comment.

srikartati commented:
LGTM

yktsubo commented:
> LGTM
> 
> There is a 2.19.3 release now, in case you want to use that.
> 
> I also suggest indicating somewhere in this document (https://github.com/vmware-tanzu/antrea/blob/master/docs/prometheus-integration.md) that Antrea-Prometheus integration is validated as part of CI using Prometheus v2.19.2 / v.2.19.3.

Thank you for your comment. I changed the version to 2.19.3.
And I also added a comment about this API in e2e README.md.

antoninbas commented on test/e2e/README.md:
```diff
@@ -120,6 +120,8 @@ enabled explicitly.
 * To run the Prometheus tests within the e2e suite, use:
 `go test -v github.com/vmware-tanzu/antrea/test/e2e --prometheus`
 
+[Target metadata API](https://prometheus.io/docs/prometheus/latest/querying/api/#querying-target-metadata) is used to validate metrics in the e2e test.
+The API has been available since Prometheus v2.4.0 but it's still experimental in v2.19.3.
```
I'm not sure this information is needed here and it may be a bit confusing. No action is required from users here, since `./infra/vagrant/push_antrea.sh --prometheus` will take care of deploying the correct version of Prometheus.

yktsubo replied:
That makes sense that this is not necessary information for users. Let me remove that change.

srikartati commented:
/test-all

srikartati commented:
As this commit has approvals, merging this. Thanks for working on this @yktsubo
